src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs

```diff
@@ -10,6 +11,7 @@
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
+    [DebuggerDisplay("Name: {Declaration.Name}, Namespace: {Declaration.Namespace}")]
```

Comment:
Looks good to me. 

Just curious, will this advance the evaluating of `Declaration`? For example, we use `Declaration` in step 10, so it will be calculated in step 10. But if in step 5 I expand the debug view to see the debugger information, `Declaration` is calculated in step 5.

Reply:
Yes, debugging might change the internal state of the object.
This is the same as you check those properties in the debugger view - which changes the state of the instance as well.
Debuggers have been optimized to avoid such cases for instance sometimes you might notice you cannot expand the values of an `IEnumerable<T>` instance because it would change the state of the instance.

Here, I believe it is safe to have them here because the Declaration of a TypeProvider is immutable and if we calculate it early, it would not cause any bad side effects